Scripts/DCS-BIOS/lib/modules/aircraft_modules/Ka-50.lua

```diff
@@ -0,0 +1,917 @@
+Ka_50:definePushButton("VOICE_MSG_REPEAT", 13, 3003, 385, "Landing Lights & Voice Warning Control Panel", "Voice message system messages repeat button")
+
+----Right Forward Panel
+--EKRAN Warning System display
+---This is an adaptation of parse_indication due to the EKRAN having arrays for some values. In most cases, parse_indication should be used.
+local function parse_EKRAN()
+	local ret = {}
+	local li = list_indication(4)
+	if li == "" then
+		return nil
+	end
+
+	local m = li:gmatch("([^\n]*)\n")
+	local newval = false
+	local name = nil
+	local value = {}
+
+	while true do
+		local line = m()
+		if not line then
+			if name ~= nil then
+				ret[name] = value
+			end
+			break
+		end
+		if line == "-----------------------------------------" then
+			newval = true
+			if name ~= nil then
+				ret[name] = value
+				name = nil
+				value = {}
+			end
+		else
+			if newval == true then
+				newval = false
+				name = line
+			else
+				value[#value + 1] = line
+			end
+		end
+	end
+	return ret
+end
+local indEKRAN = nil
+Ka_50:addExportHook(function()
+	indEKRAN = parse_EKRAN()
+end)
+local function getEKRAN_memory()
+	if indEKRAN == nil or indEKRAN.txt_memory == nil then
+		return "0"
+	end
+	return "1"
+end
+local function getEKRAN_queue()
+	if indEKRAN == nil or indEKRAN.txt_queue == nil then
+		return "0"
+	end
+	return "1"
+end
+local function getEKRAN_failure()
+	if indEKRAN == nil or indEKRAN.txt_failure == nil then
+		return "0"
+	end
+	return "1"
+end
+Ka_50:defineString("EKRAN_MEMORY", getEKRAN_memory, 1, "EKRAN", "Memory message")
+Ka_50:defineString("EKRAN_QUEUE", getEKRAN_queue, 1, "EKRAN", "Queue message")
+Ka_50:defineString("EKRAN_FAILURE", getEKRAN_failure, 1, "EKRAN", "Failure message")
+local function getEKRAN_txt1_line1()
+	if indEKRAN == nil or indEKRAN.txt_1 == nil then
+		return ""
+	end
+	return indEKRAN.txt_1[1] or ""
+end
+local function getEKRAN_txt1_line2()
+	if indEKRAN == nil or indEKRAN.txt_1 == nil then
+		return ""
+	end
+	return indEKRAN.txt_1[2] or ""
+end
+local function getEKRAN_txt1_line3()
+	if indEKRAN == nil or indEKRAN.txt_1 == nil then
+		return ""
+	end
+	return indEKRAN.txt_1[3] or ""
+end
+local function getEKRAN_txt1_line4()
+	if indEKRAN == nil or indEKRAN.txt_1 == nil then
+		return ""
+	end
+	return indEKRAN.txt_1[4] or ""
+end
+Ka_50:defineString("EKRAN_TXT1_LINE1", getEKRAN_txt1_line1, 10, "EKRAN", "EKRAN txt 1 line 1")
+Ka_50:defineString("EKRAN_TXT1_LINE2", getEKRAN_txt1_line2, 10, "EKRAN", "EKRAN txt 1 line 2")
+Ka_50:defineString("EKRAN_TXT1_LINE3", getEKRAN_txt1_line3, 10, "EKRAN", "EKRAN txt 1 line 3")
+Ka_50:defineString("EKRAN_TXT1_LINE4", getEKRAN_txt1_line4, 10, "EKRAN", "EKRAN txt 1 line 4")
+local function getEKRAN_txt2_line1()
+	if indEKRAN == nil or indEKRAN.txt_2 == nil then
+		return ""
+	end
+	return indEKRAN.txt_2[1] or "          "
```

Comment:
These return values don't need strings full of spaces either. There are a lot like this that should probably fixed. Maybe using coerce_nil_to_string is best after all so nobody has to think about it? 
```suggestion
	return indEKRAN.txt_2[1] or ""
```